docs/notebooks/openai_gym_lunar_lander.pct.py

```diff
@@ -0,0 +1,244 @@
+# %% [markdown]
+# # Trieste meets OpenAI Gym
+#
+# This notebook demonstrates how to use Trieste to apply Bayesian optimization to a problem that is slightly more practical than classical optimization benchmarks shown used in other tutorials. We will use OpenAI Gym, which is a popular toolkit for reinforcement learning (RL) algorithms.
+#
+# Concretely, we are going to take the [Lunar Lander](https://gym.openai.com/envs/LunarLander-v2/) environment, define a search space and describe it as an optimization problem, and use Trieste to find an optimal solution for the problem. And hopefully avoid too many landers crashing on the Moon surface along the way.
+
+# %%
+import tensorflow as tf
+import numpy as np
+import trieste
+import gpflow
+
+
+import gym
+env_name = "LunarLander-v2"
+env = gym.make(env_name)
+
+seed = 1793
+
+np.random.seed(seed)
+tf.random.set_seed(seed)
+env.seed(seed)
+
+
+# %% [markdown]
+# ### Introduction
+#
+# Let's start by discussing the problem itself. In the Lunar Lander environment we are controlling a space module that needs to land on a Moon surface. The surface is piecewise linear and is generated randomly, but always has a flat landing pad in the middle (marked with flags on the renders). The module starts at the top with some random initial speed and direction. We are controlling three engines on the module: one on each side, and one at the bottom. At each step of the simulation we can choose to either fire one of the engines, or do nothing. The ultimate goal of the simulation is to land safely on the marked landing pad.
```

Comment:
remove first sentence?

Reply:
Why? Does it hurt?

Reply:
sorry, just the old habit of chasing the superfluous. Keep it if you like it, this is not an 8-page NeurIPS paper

Reply:
will keep, thanks!